connect/runtime/src/test/java/org/apache/kafka/connect/integration/RestExtensionIntegrationTest.java

```diff
@@ -70,10 +83,81 @@ public void testImmediateRequestForListOfConnectors() throws IOException, Interr
         );
     }
 
+    @Test
+    public void testConnectorHealthAndConfig() throws IOException, InterruptedException {
+        // setup Connect worker properties
+        Map<String, String> workerProps = new HashMap<>();
+        workerProps.put(REST_EXTENSION_CLASSES_CONFIG, IntegrationTestRestExtension.class.getName());
+
+        // build a Connect cluster backed by Kafka and Zk
+        connect = new EmbeddedConnectCluster.Builder()
+            .name("connect-cluster")
+            .numWorkers(1)
+            .numBrokers(1)
+            .workerProps(workerProps)
+            .build();
+
+        // start the clusters
+        connect.start();
+
+        WorkerHandle worker = connect.workers().stream()
+            .findFirst()
+            .orElseThrow(() -> new AssertionError("At least one worker handle should be available"));
+
+        waitForCondition(
+            this::extensionIsRegistered,
+            REST_EXTENSION_REGISTRATION_TIMEOUT_MS,
+            "REST extension was never registered"
+        );
```

Comment:
I see that `testImmediateRequestForListOfConnectors` does almost the same thing. Can we perhaps refactor this?